internal/debug/api.go

```diff
@@ -189,10 +191,44 @@ func (*HandlerT) WriteMemProfile(file string) error {
 	return writeProfile("heap", file)
 }
 
-// Stacks returns a printed representation of the stacks of all goroutines.
-func (*HandlerT) Stacks() string {
+// Stacks returns a printed representation of the stacks of all goroutines. It
+// also permits the following optional filters to be used:
+//   - filter: boolean expression of packages to filter for
+func (*HandlerT) Stacks(filter *string) string {
 	buf := new(bytes.Buffer)
 	pprof.Lookup("goroutine").WriteTo(buf, 2)
+
+	// If any filtering was requested, execute them now
+	if filter != nil && len(*filter) > 0 {
+		expanded := *filter
+
+		// The input filter is a logical expression of package names. Transform
+		// it into a proper boolean expression that can be fed into a parser and
+		// interpreter:
+		//
+		// E.g. (eth || snap) && !p2p -> (eth in Value || snap in Value) && p2p not in Value
+		expanded = regexp.MustCompile("[/\\.A-Za-z0-9_-]+").ReplaceAllString(expanded, "`$0` in Value")
+		expanded = regexp.MustCompile("!(`[/\\.A-Za-z0-9_-]+`)").ReplaceAllString(expanded, "$1 not")
```

Comment:
```suggestion
		expanded = regexp.MustCompile("[/\\.A-Za-z0-9_-:]+").ReplaceAllString(expanded, "`$0` in Value")
		expanded = regexp.MustCompile("!(`[/\\.A-Za-z0-9_-:]+`)").ReplaceAllString(expanded, "$1 not")
```

For 
```
> debug.stacks("console.go:419")
INFO [09-20|13:22:16.459] Expanded filter expression               filter=console.go:419 expanded="`console.go:419` in Value"
goroutine 140 [chan receive]:
github.com/ethereum/go-ethereum/console.(*Console).readLines(0xc000596310, 0xc0004566e8, 0xc000dc3ff0, 0xc0000bb7b0)
	github.com/ethereum/go-ethereum/console/console.go:419 +0x46
created by github.com/ethereum/go-ethereum/console.(*Console).Interactive
	github.com/ethereum/go-ethereum/console/console.go:363 +0x1ef

```

Reply:
Just move the `:` to the front of the regexp. I think the `-` must be the last, otherwise it's interpreted as an interval i.e. from `_` to `:`

Reply:
Done. THat's actually how I had it in my own code, but when I wrote the comment here I moved it. WCGW right? ;)